link/kprobe.go

```diff
@@ -146,11 +147,9 @@ func pmuKprobe(symbol string, ret bool) (*perfEvent, error) {
 
 	// Kernel has perf_kprobe PMU available, initialize perf event.
 	return &perfEvent{
-		fd:       internal.NewFD(uint32(fd)),
```

Comment:
`perfEvent` should always have the appropriate event type set out of the gate. This avoids `Kprobe` or `Kretprobe` becoming tightly coupled with their internal functions.